tests/integ/test_basic.py

```diff
@@ -36,6 +40,110 @@ def test_crud(engine):
     assert [same_user] == excinfo.value.objects
 
 
+def test_model_defaults(engine):
+    class ColumnDefaultsModel(BaseModel):
+        hash = Column(Integer, hash_key=True, default=12)
+        range = Column(Integer, range_key=True, default=24)
+        other = Column(Integer, default=48)
+    engine.bind(ColumnDefaultsModel)
+
+    obj = ColumnDefaultsModel()
+    assert obj.hash == 12
+    assert obj.range == 24
+    assert obj.other == 48
+
+    engine.save(obj)
+
+    same_obj = ColumnDefaultsModel(hash=12, range=24)
+    engine.load(same_obj)
+
+    assert same_obj.hash == 12
+    assert same_obj.range == 24
+    assert same_obj.other == 48
+
+
+def test_model_defaults_load(engine):
+    class ColumnDefaultLoadModel(BaseModel):
+        hash = Column(Integer, hash_key=True, default=12)
+        range = Column(Integer, range_key=True, default=24)
+        other = Column(Integer, default=48)
+    engine.bind(ColumnDefaultLoadModel)
+
+    obj = ColumnDefaultLoadModel(hash=333, range=333)
+    engine.save(obj)
+
+    same_obj = ColumnDefaultLoadModel(hash=obj.hash, range=obj.range)
+    engine.load(same_obj)
+
+    assert same_obj.hash == 333
+    assert same_obj.range == 333
+    assert same_obj.other == 48
+
+
+def test_model_default_func(engine):
+    def get_int():
+        return 404
+
+    def random_int():
+        return randint(1, 100)
+
+    class ColumnDefaultFuncModel(BaseModel):
+        hash = Column(UUID, hash_key=True, default=uuid.uuid4())
+        range = Column(Integer, range_key=True, default=get_int)
+        other = Column(Integer, default=random_int)
+    engine.bind(ColumnDefaultFuncModel)
+
+    obj = ColumnDefaultFuncModel()
+    assert isinstance(obj.hash, uuid.UUID)
+    assert obj.range == 404
+
+    engine.save(obj)
+
+    same_obj = ColumnDefaultFuncModel(hash=obj.hash, range=obj.range)
+    engine.load(same_obj)
+
+    assert same_obj.hash == obj.hash
+    assert same_obj.range == obj.range
+    assert same_obj.other == obj.other
+
+
+def test_model_default_projection(engine):
```

Comment:
I've added another test in the dev-2.0 branch that expands the contract for partial loads beyond not calling the default function; when you merge that branch in I expect it will cause some problems.  The goal is to ensure saving a partially-loaded object does not delete or null out fields it simply doesn't know about; it should only send updates for fields it knows have changed.

In this test, you could add a column `extra = Column(String)`, set it on the initial instance `instance = MyModel(..., extra="foo")` and then simply call `engine.save(same_instance)` right after the `assert same_instance.password is None`.